protoc-gen-grpc-gateway/descriptor/types.go

```diff
@@ -249,15 +249,37 @@ func (p FieldPath) AssignableExpr(msgExpr string) string {
 	if l == 0 {
 		return msgExpr
 	}
-	components := []string{msgExpr}
+
+	prepare := []string{}
+	fqnm := msgExpr
```

Comment:
Maybe you intended "fqmn"?  Also I don't think it is a "fully-qualified *message* name" in a proto schema but it was something in the generated golang code.

I wonder why you needed to replace `components` with it.

Reply:
I just have not found a word to name it, it is a RHS to assign.

**components** is replaced with **prepare** variable, which is joined at last like the **components**. the **fqnm** is used directly at last , in each iteration, it is accumulated as a partly qualified field name. i.e.

protoMsg
protoMsg.A
protoMsg.A.B.(M_F).F
protoMsg.A.B.(M_F).F.C